src/Symfony/Bridge/Twig/Tests/Extension/HttpFoundationExtensionTest.php

```diff
@@ -43,6 +43,21 @@ public function getGenerateAbsoluteUrlData()
         );
     }
 
+    public function testGenerateAbsoluteUrlWithScriptFileName()
+    {
+        $request = Request::create('http://localhost/app/web/app_dev.php');
+        $request->server->set('SCRIPT_FILENAME', '/var/www/app/web/app_dev.php');
```

Comment:
Note that we need a non empty `SCRIPT_FILENAME` otherwise the base url is empty and the current behavior works for that particular case.